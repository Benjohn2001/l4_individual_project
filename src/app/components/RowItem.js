import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, View } from "react-native";
import { Feather } from "@expo/vector-icons";

function RowItem(props) {
  const { title, onPress, icon, color } = props;
  return (
    <View className="pt-3 w-full items-center">
      <TouchableOpacity
        onPress={onPress}
        className="bg-secondaryPurple  h-14 w-11/12 pt-4 rounded-2xl mx-1.5"
      >
        <View className="flex-row">
          <View className="ml-5">
            <Feather name={icon} size={20} color={color} />
          </View>
          <Text className="text-black text-sm text-center ml-10">{title}</Text>
          <View className="ml-auto mr-5">
            <Feather name="arrow-right" size={20} color="black" />
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
}

RowItem.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
};

export default RowItem;
