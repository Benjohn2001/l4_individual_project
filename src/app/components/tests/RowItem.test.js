import RowItem from '../RowItem';
import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";

test('Check RowItem renders correctly and onpress works',()=>{
    const onPressMock = jest.fn();
    render(
        <RowItem 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
            icon="activity"
            color="blue"
        />,
      );
      fireEvent.press(screen.getByText('press'));
      expect(onPressMock)
      const domTree = renderer.create(
        <RowItem 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
            icon="activity"
            color="blue"
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})