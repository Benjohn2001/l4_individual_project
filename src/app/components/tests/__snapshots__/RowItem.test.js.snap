// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check RowItem renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E7E7FF",
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "height": 56,
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
        "paddingTop": 16,
        "width": "91.666667%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 20,
            },
          ]
        }
      >
        <Text />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontSize": 14,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "marginLeft": 40,
            },
          ]
        }
      >
        press
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": "auto",
            },
            Object {
              "marginRight": 20,
            },
          ]
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
